### PR TITLE
Fixes #1903: Fix image scaling in navigation drawer

### LINKED ISSUE #1903: Background image stretched in Android app sidebar
### Actual behaviour

I have manually set my background image, it's 1920×1920 (ie. 1:1 aspect). When I open the Android sidebar, it appears stretched horizontally. It appears normally on the web interface login.

### Expected behaviour

The Android sidebar image should have the same aspect as the original image (cropped or scaled I don't care).
 
### Steps to reproduce
1. Set a square image for the background via Admin > Theming. 
2. Open and refresh the Android app.

### Environment data
Android version: 7.1.2

Device model: HTC One M9

Stock or customized system: Custom (LineageOS 14.1)

Nextcloud app version: 2.0.0 (beta version)

Nextcloud server version: 12.0.3

I haven't attached logs because I don't have access to the server logs right this minute, and the Android app doesn't log. However, here's a couple of images, first the original scaled down (just for reference) then a screenshot of the Android app.

bg1

bg2

### PR DESCRIPTION
**Pull request by przybylski**
Fix #1903

* Insted of setting a background in ViewGroup add a ImageView inside it, stretch it, set cropping scaleType and then set a background image downloaded from server.
* CenterCrop will scale the image to fill the View, if needed image will be cropped in the center but the aspect ration will be kept
* Cleanup code to use more general types instead of direct layout type, previously enforcing LinearLayout wasn't using anything special about this class.

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
Welcome back and thank you for this PR 🎉 

Sample image with aspect ratio:

**Comment by tobiasKaminsky** [comment]
@przybylski can you do the same on user info screen (drawer -> manage accounts -> select one account)? 🎉

File: src/main/res/layout/drawer_header.xml [hunk 1]
```diff
@@ -129,4 +135,4 @@
 
     </RelativeLayout>
 
-</LinearLayout>
\ No newline at end of file
+</FrameLayout>
```
Comment (mario) [inline src/main/res/layout/drawer_header.xml]: New line please :)

**Comment by mario** [comment]
Looks nice, just one minor comment regarding new line.

Glad to have you here!

**Comment by przybylski** [comment]
@tobiasKaminsky since Account Info has a little different approach on how header is build I changed the behaviour a little bit. it is no longer snapping on scroll.

**Comment by tobiasKaminsky** [comment]
Thank you for the quick response :-)
I am fine with this. @AndyScherzinger implemented the snapping header. So let us wait for his opinion.

**Comment by mario** [comment]
@tobiasKaminsky merge now, think later? :P Otherwise this won't be in 2.1.

**Comment by tobiasKaminsky** [comment]
Right, @AndyScherzinger can change the behaviour if he wants to later :-)
Please approve :-)

**Comment by mario** [comment]
👍

**Comment by mario** [comment]
Done.

**Comment by tobiasKaminsky** [comment]
Merging with only one approval as @AndyScherzinger is on vacation.

**Comment by AndyScherzinger** [comment]
Totally fine with me!a

🎉 Welcome back @przybylski 🎉

**Pull request merged**